shell/platform/fuchsia/flutter/vulkan_surface.cc

```diff
@@ -226,57 +228,112 @@ bool VulkanSurface::CreateFences() {
   return true;
 }
 
-bool VulkanSurface::AllocateDeviceMemory(sk_sp<GrDirectContext> context,
-                                         const SkISize& size,
-                                         zx::vmo& exported_vmo) {
+bool VulkanSurface::AllocateDeviceMemory(
+    fuchsia::sysmem::AllocatorSyncPtr sysmem_allocator,
+    sk_sp<GrDirectContext> context,
+    const SkISize& size) {
   if (size.isEmpty()) {
     return false;
   }
 
-  VulkanImage vulkan_image;
-  if (!CreateVulkanImage(vulkan_provider_, size, &vulkan_image)) {
-    FML_DLOG(ERROR) << "Failed to create VkImage";
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr local_token;
+  zx_status_t status =
+      sysmem_allocator->AllocateSharedCollection(local_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to allocate collection");
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr scenic_token;
+  status = local_token->Duplicate(std::numeric_limits<uint32_t>::max(),
+                                  scenic_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to duplicate token");
+  status = local_token->Sync();
+  LOG_AND_RETURN(status != ZX_OK, "Failed to sync token");
+  fuchsia::sysmem::BufferCollectionTokenSyncPtr vulkan_token;
+  status = local_token->Duplicate(std::numeric_limits<uint32_t>::max(),
+                                  vulkan_token.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to duplicate token");
+  status = local_token->Sync();
+  LOG_AND_RETURN(status != ZX_OK, "Failed to sync token");
+
+  session_->RegisterBufferCollection(buffer_id_, std::move(scenic_token));
+
+  fuchsia::sysmem::BufferCollectionSyncPtr buffer_collection;
+  status = sysmem_allocator->BindSharedCollection(
+      std::move(local_token), buffer_collection.NewRequest());
+  LOG_AND_RETURN(status != ZX_OK, "Failed to bind collection");
+
+  fuchsia::sysmem::BufferCollectionConstraints constraints;
+  constraints.min_buffer_count = 1;
+  constraints.usage.vulkan =
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_COLOR_ATTACHMENT |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_TRANSFER_DST |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_TRANSFER_SRC |
+      fuchsia::sysmem::VULKAN_IMAGE_USAGE_SAMPLED;
```

Comment:
Do we need these? It used to be that we needed it to match scenic but now that we're using sysmem it's enough to set the usage flutter needs. This would also require changing line 74.

Reply:
What usage does Flutter need?

Reply:
Don't know. We used to need all these to match the usage in Scenic but that's no longer needed when using sysmem. Let's keep them as is but put a comment here about removing usages that are not required by Flutter.